crates/storage/src/tests/common.rs

```diff
@@ -0,0 +1,30 @@
+use calimero_store::config::StoreConfig;
+use calimero_store::db::RocksDB;
+use calimero_store::Store;
+use tempfile::{tempdir, TempDir};
+
+/// Creates a new temporary store for testing.
+///
+/// This function creates a new temporary directory and opens a new store in it,
+/// returning the store and the directory. The directory and its test data will
+/// be cleaned up when the test completes.
+///
+pub fn create_test_store() -> (Store, TempDir) {
+    // Note: It would be nice to have a way to test against an in-memory store,
+    // but InMemoryDB is not integrated with Store and there's currently no way
+    // to support both. It may be that the Database trait is later applied to
+    // InMemoryDB as well as RocksDB, in which case the storage Interface could
+    // be changed to work against Database implementations and not just Store.
```

Comment:
this is the case, though:

https://github.com/calimero-network/core/blob/8ecf05ae597ef4ded21a1edd664ee482e16e03de/crates/store/src/db/memory.rs#L132

https://github.com/calimero-network/core/blob/8ecf05ae597ef4ded21a1edd664ee482e16e03de/crates/store/src/tests/db/memory.rs